libsvm-format/java-formats/libsvm/src/com/dataiku/dss/formats/libsvm/LIBSVMFormat.java

```diff
@@ -0,0 +1,261 @@
+                lineNumber++;
+                labelMatcher.reset(line);
+                tokenMatcher.reset(line);
+
+                // If no label is found the line is considered as invalid and skipped
+                if (!labelMatcher.find()) {
+                    limitedLogger.logV("Line %d is invalid (no label detected): %s", lineNumber, line);
+                    continue;
+                }
+                Row row = rf.row();
+
+                String label = labelMatcher.group(1);
+                row.put(labelColumn, label);
+
+                int start = labelMatcher.end();
+                tokenMatcher.region(start, line.length());
+
+                while (tokenMatcher.find()) {
+                    String index = tokenMatcher.group(1);
+                    String value = tokenMatcher.group(2);
+
+                    Column column = columns.get(index);
+                    // Creating the newly found column if possible
+                    if (column == null) {
+                        if (columns.size() > maxFeatures) continue;
+
+                        column = cf.column(index);
+                        columns.put(index, column);
+                    }
+
+                    start = tokenMatcher.end();
+                    tokenMatcher.region(start, line.length());
+
+                    row.put(column, value);
+                }
+
+                if (!tokenMatcher.hitEnd()) {
+                    limitedLogger.logV("Invalid end-of-line at [%d,%d]: %s", lineNumber, start, line.substring(start));
+                }
+
+                out.emitRow(row);
+            }
+            out.lastRowEmitted();
+        }
+
+        private void parseIntoSingleJSONColumn(ProcessorOutput out, ColumnFactory cf, RowFactory rf,
+                                               BufferedReader bf, LimitedLogContext limitedLogger) throws Exception {
+            Column labelColumn = cf.column("Label");
+            Column featuresColumn = cf.column("Features");
+
+            StringBuilder features = new StringBuilder();
+            Matcher labelMatcher = LABEL_PATTERN.matcher("");
+            Matcher tokenMatcher = TOKEN_PATTERN.matcher("");
+
+            String line;
+            int lineNumber = 0;
+
+            while ((line = bf.readLine()) != null) {
+                lineNumber++;
+                labelMatcher.reset(line);
+                tokenMatcher.reset(line);
+
+                // If no label is found the line is considered as invalid and skipped
+                if (!labelMatcher.find()) {
+                    limitedLogger.logV("Line %d is invalid (no label detected): %s", lineNumber, line);
+                    continue;
+                }
+                Row row = rf.row();
+
+                String label = labelMatcher.group(1);
+                row.put(labelColumn, label);
+
+                int start = labelMatcher.end();
+                tokenMatcher.region(start, line.length());
```

Comment:
[Nitpick] Could factorize the code of the 2 methods up to this line

Reply:
`start` is used in error messages